### DISCUSSION TITLE
Lsposed with Graphene OS 13

### DISCUSSION THREAD

**Discussion by kushmittal2009** [LSPosed/LSPosed#2474] [General]
There are many issues reported with Lsposed not being supported with graphene OS. Is there any way to make it supported?

**Comment by yujincheng08** [comment]
If you choose a security ROM like Graphene OS, you'd better not root your device.

**Comment by Aeroverra** [comment]
I am cool with not maintaining it for Graphene but I disagree with the broad assumption that no one should root their device and cannot do so in a safe manner.  Isn't that the whole foundation of what this is build on?

**Comment by BrandonSLi** [comment]
For some people, rooting is, in a sense, necessary for security. For example, individuals living under oppressive regimes can only gain access to the cellular network by rooting their GrapheneOS or CalyxOS phones. Otherwise, they would be limited to using their country's notorious cellphones instead of a Pixel.

**Comment by aviraxp** [comment]
GrapheneOS heavily modified libc and zygote forking which makes lsposed hooks hard to implement. We have no plan on fixing this.

**Reply by znbaboy** [reply]
please reconsider this
